skyportal/facility_apis/lt.py

```diff
@@ -0,0 +1,855 @@
+    _observation_types = ['H']
+    _exposure_types = ['1x120s', '2x150s']
+
+    form_json_schema = {
+        "type": "object",
+        "properties": {
+            "observation_type": {
+                "type": "string",
+                "enum": _observation_types,
+                "default": "H",
+            },
+            "exposure_type": {
+                "type": "string",
+                "enum": _exposure_types,
+                "default": '1x120s',
+            },
+            "priority": {"type": "string", "enum": ["1", "5"], "default": "1"},
+            "LT_proposalID": {"type": "string"},
+            "start_date": {
+                "type": "string",
+                "format": "date",
+                "default": datetime.utcnow().date().isoformat(),
+                "title": "Start Date (UT)",
+            },
+            "end_date": {
+                "type": "string",
+                "format": "date",
+                "title": "End Date (UT)",
+                "default": (datetime.utcnow().date() + timedelta(days=7)).isoformat(),
+            },
+            "maximum_airmass": {
+                "title": "Maximum Airmass (1-3)",
+                "type": "number",
+                "default": 2.0,
+            },
+            "maximum_seeing": {
+                "title": "Maximum Seeing [arcsec] (0-5)",
+                "type": "number",
+                "default": 1.2,
+            },
+            "sky_brightness": {
+                "title": "Maximum allowable Sky Brightness, Dark + X magnitudes [arcsec] (0-5)",
+                "type": "number",
+                "default": 2.0,
+            },
+            "photometric": {
+                "title": "Does this observation require photometric conditions?",
+                "type": "boolean",
+            },
+            "binning": {"type": "string", "enum": ["1x1", "2x2"], "default": "1x1"},
+        },
+        "required": [
+            "observation_type",
+            "exposure_type",
+            "priority",
+            "start_date",
+            "end_date",
+            "maximum_airmass",
+            "maximum_seeing",
+            "binning",
+        ],
+    }
+
+    ui_json_schema = {}
+
+
+class SPRATAPI(LTAPI):
+
+    """An interface to LT SPRAT operations."""
+
+    # subclasses *must* implement the method below
+    @staticmethod
+    def submit(request):
+
+        """Submit a follow-up request to LT's SPRAT.
+
+        Parameters
+        ----------
+        request: skyportal.models.FollowupRequest
+            The request to add to the queue and the SkyPortal database.
+        """
+
+        from ..models import DBSession, FacilityTransaction
+
+        altdata = request.allocation.load_altdata()
+        if not altdata:
+            return
+
+        ltreq = SPRATRequest()
+        observation_payload = ltreq._build_prolog()
+        ltreq._build_project(observation_payload, request)
+        ltreq._build_inst_schedule(observation_payload, request)
```

Comment:
Can this pattern be simplified to 

```
ltreq = SPRATRequest(request)
```

Reply:
still wonder about the above. Can't these calls to private methods be moved to the constructor of the Request objects ?

Reply:
Sorry can you help me by fleshing this out a bit?